aten/src/ATen/core/ivalue.cpp

```diff
@@ -463,6 +463,52 @@ std::ostream& IValue::repr(
   }
 }
 
+bool operator<(const ivalue::Tuple& lhs, const ivalue::Tuple& rhs) {
+  const auto& lhs_vals = lhs.elements();
+  const auto& rhs_vals = rhs.elements();
+
+  // Comparing tuples with different length is undefined.
+  TORCH_INTERNAL_ASSERT(lhs_vals.size() == rhs_vals.size());
+
+  const size_t n = lhs_vals.size();
+  for(size_t i = 0; i < n; ++i) {
+    if(lhs_vals[i] < rhs_vals[i]) {
+      return true;
+    }
+    if(lhs_vals[i] > rhs_vals[i]) {
+      return false;
+    }
+  }
+
+  // Reaching here means rhs and lhs are equal.
+  return false;
+}
+
+bool operator<(const IValue& a, const IValue& b) {
+  // Must be same type to be compared.
+  TORCH_INTERNAL_ASSERT(a.tag == b.tag);
+  switch (a.tag) {
+    case IValue::Tag::Tensor:
+       return a.toTensor().lt(b.toTensor()).is_nonzero();
+    case IValue::Tag::Double:
+       return a.toDouble() < b.toDouble();
+    case IValue::Tag::Int:
+       return a.toInt() < b.toInt();
+    case IValue::Tag::Bool:
+       return a.toBool() == false && b.toBool() == true;
```

Comment:
Maybe the way you've written is more readable. I don't feel too strongly about this.

```suggestion
       return !a.toBool() && b.toBool();
```

Reply:
Ack. Showing explicit values is indeed slightly more readable like you said.